### ISSUE TITLE
Replace jsonschema.RefResolver as is being deprecated

### ISSUE THREAD

**Issue by martinhoyer** [teemtee/fmf#194]
Copy-pasting the deprecation warning:

`fmf/base.py:682: DeprecationWarning: jsonschema.RefResolver is deprecated as of v4.18.0, in favor of the https://github.com/python-jsonschema/referencing library, which provides more compliant referencing behavior as well as more flexible APIs for customization. A future release will remove RefResolver. Please file a feature request (on referencing) if you are missing an API for the kind of customization you need.`

4.18.0 was released July 6th.

If there is no-one available to work on this now, perhaps a max version lock in installrequires like `jsonschema<4.18` would be a adequate temporary solution.

**Comment by lukaszachy** [comment]
Hm, while locking version will work for PyPI we can't do that in fedora-rawhide which might get 4.18 at some point.

**Comment by martinhoyer** [comment]
> Hm, while locking version will work for PyPI we can't do that in fedora-rawhide which might get 4.18 at some point.

Indeed, see [bz#2223179](https://bugzilla.redhat.com/show_bug.cgi?id=2223179))

Locking the version now would ensure parity between rpms and pypi, but is not crucial I guess.
